analytics/src/test/java/com/segment/analytics/QueueFileTest.kt

```diff
@@ -0,0 +1,953 @@
+            Assertions.assertThat(ex)
+                .hasMessage("File is corrupt; last position stored in header (-2147483648) is invalid.")
+        }
+    }
+
+    @Test
+    @Throws(IOException::class)
+    fun removeMultipleDoesNotCorrupt() {
+        var queue = QueueFile(file)
+        for (i in 0 until 10) {
+            queue.add(values[i])
+        }
+
+        queue.remove(1)
+        Assertions.assertThat(queue.size()).isEqualTo(9)
+        Assertions.assertThat(queue.peek()).isEqualTo(values[1])
+
+        queue.remove(3)
+        queue = QueueFile(file)
+        Assertions.assertThat(queue.size()).isEqualTo(6)
+        Assertions.assertThat(queue.peek()).isEqualTo(values[4])
+
+        queue.remove(6)
+        Assertions.assertThat(queue.isEmpty).isTrue()
+        Assertions.assertThat(queue.peek()).isNull()
+    }
+
+    @Test
+    @Throws(IOException::class)
+    fun removeDoesNotCorrupt() {
+        var queue = QueueFile(file)
+        queue.add(values[127])
+        val secondStuff = values[253]
+        queue.add(secondStuff)
+        queue.remove()
+
+        queue = QueueFile(file)
+        Assertions.assertThat(queue.peek()).isEqualTo(secondStuff)
+    }
+
+    @Test
+    @Throws(IOException::class)
+    fun removeFromEmptyFileThrows() {
+        val queue = QueueFile(file)
+
+        try {
+            queue.remove()
+            Assertions.fail("Should have thrown about removing from empty file.")
+        } catch (ignored: NoSuchElementException) {
+        }
+    }
+
+    @Test
+    @Throws(IOException::class)
+    fun removeNegativeNumberOfElementsThrows() {
+        val queue = QueueFile(file)
+        queue.add(values[127])
+
+        try {
+            queue.remove(-1)
+            Assert.fail("Should have thrown about removing negative number of elements.")
+        } catch (ex: IllegalArgumentException) {
+            Assertions.assertThat(ex)
+                .hasMessage("Cannot remove negative (-1) number of elements.")
+        }
+    }
+
+    @Test
+    @Throws(IOException::class)
+    fun removeZeroElementsDoesNothing() {
+        val queue = QueueFile(file)
+        queue.add(values[127])
+
+        queue.remove(0)
+        Assertions.assertThat(queue.size()).isEqualTo(1)
+    }
+
+    @Test
+    @Throws(IOException::class)
+    fun removeBeyondQueueSizeElementsThrows() {
+        val queue = QueueFile(file)
+        queue.add(values[127])
+
+        try {
+            queue.remove(10)
+            Assertions.fail("Should have thrown about removing too many elements.")
+        } catch (ex: java.lang.IllegalArgumentException) {
+            Assertions.assertThat(ex)
+                .hasMessage("Cannot remove more elements (10) than present in queue (1).")
+        }
+    }
+
+    @Test
+    @Throws(IOException::class)
+    fun removingBigDamnBlocksErasesEffectively() {
+        val bigBoy = ByteArray(7000)
+        for (i in 0 until 7000 step 100) {
+            values[100]!!.let {
```

Comment:
to improve readability, can we name the `it` and `it1` to something more apt here? something like `val` and `valSize`